### PR TITLE
Add EfficientNet model support

### LINKED ISSUE #143: [Classification Model] Please add EfficientNet model family
The EfficientNet model is a classification model based on the [EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks](https://arxiv.org/abs/1905.11946) paper.

Please implement

- [ ] EfficientNet B0
- [ ] EfficientNet B1
- [ ] EfficientNet B2
- [ ] EfficientNet B3
- [ ] EfficientNet B4
- [ ] EfficientNet B5
- [ ] EfficientNet B6
- [ ] EfficientNet B7

with IMAGENET1K_V1 or IMAGENET1K_V2 pretrained weights when available.

### LINKED ISSUE #144: [Classification Model] Please add EfficienNet_v2 model family
The EfficientNet_v2 model are classification models from [EfficientNetV2: Smaller Models and Faster Training](https://arxiv.org/abs/2104.00298). paper.

Please implement

- [ ] EfficientNet-v2-S
- [ ] EfficientNet-v2-M
- [ ] EfficientNet-v2-L

with pretrained weights.

### PR DESCRIPTION
**Pull request by koshtiakanksha**
This PR adds initial support for EfficientNet variants B0–B7, including architecture, config functions, and test coverage.

Currently waiting for pretrained weights. Working on converting .pth files and ensuring compatibility with R's torch::load_state_dict().

Closes #143 

Closes #144

### CODE REVIEW

**Comment by koshtiakanksha** [comment]
Hi @cregouby. I’ve implemented all the EfficientNet B0–B7 models with support for loading PyTorch weights. I wasn’t sure how to upload the .pth files to the cloud for automatic download, so for now, I tested it using local paths to verify the implementation. Everything seems to work across all variants. Could you please review the code and guide me on how to properly upload the weights to the expected storage location? Let me know if any changes are needed.

File: tests/testthat/test-models-efficientnet.R
Comment (cregouby) [inline tests/testthat/test-models-efficientnet.R:1]: **todo** Could we test the models with their native image size (i.e. according to the documented `size` parameter ?
**todo** Could we add a test with a non-default `num_classes` ?

**Review by cregouby — CHANGES_REQUESTED** [review]
**praise** much clearer model documentation, thanks ! Will be used as reference !

File: R/models-efficientnet.R [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  args <- rlang::list2(...)
+  model <- do.call(efficientnet, append(args, list(
+    width_coefficient = width,
+    depth_coefficient = depth,
+    dropout = dropout
+  )))
+
+  if (pretrained) {
+    local_path <- here::here("tools", "models", paste0(arch, ".pth"))
+    message("Using local weights: ", local_path)
```
Comment (cregouby) [inline R/models-efficientnet.R:217]: **todo** please stick to messaging functions in `condition.R` file so that they get translated.

File: R/models-efficientnetv2.R [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+  model <- do.call(efficientnet_v2, append(args, list(
+    cfgs = cfgs,
+    dropout = dropout,
+    firstconv_out = firstconv_out
+  )))
+
+  if (pretrained) {
+    local_path <- here::here("tools", "models", paste0(arch, ".pth"))
+    print(local_path)
```
Comment (cregouby) [inline R/models-efficientnetv2.R:195]: **todo** please stick to messaging functions in `condition.R` file so that they get translated.

**Review by cregouby — CHANGES_REQUESTED** [review]
**todo** as the models are now available online, you should switch to the online version.

**Comment by koshtiakanksha** [comment]
Hi @tdhock I have documented only num_classes explicitly, since that's the only argument shown in PyTorch's documentation. Other parameters like dropout, norm_layer, and others are used internally in our implementation but aren't documented, just like in PyTorch. Currently, we're inheriting from model_resnet18, which places num_classes after the ... argument, which feels a bit unnatural. Should we switch to writing all parameters explicitly instead, to better control the order and match PyTorch more closely? Or is the current approach okay?

**Comment by cregouby** [comment]
> Hi @tdhock I have documented only num_classes explicitly, since that's the only argument shown in PyTorch's documentation. Other parameters like dropout, norm_layer, and others are used internally in our implementation but aren't documented, just like in PyTorch. Currently, we're inheriting from model_resnet18, which places num_classes after the ... argument, which feels a bit unnatural. Should we switch to writing all parameters explicitly instead, to better control the order and match PyTorch more closely? Or is the current approach okay?

Hello @koshtiakanksha, 

Unfortunately, you cannot add a parameter out of the wild (see https://github.com/mlverse/torchvision/actions/runs/16336647880/job/46150010417?pr=166). I would advise you to put the num_classes information as part of the `...` parameter description.
